lib/propmangle.js

```diff
@@ -150,6 +150,11 @@ function mangle_properties(ast, options) {
             if (!(keep_quoted && node.quote))
                 node.key = mangle(node.key);
         }
+        else if (node instanceof AST_ConciseMethod && node.name && node.key.name) {
```

Comment:
The code in #2262 is indeed simpler/better.

Sometimes mistakes are introduced in code and over time shrines are built around them and the point of the code is forgotten. When in doubt just remove the code.